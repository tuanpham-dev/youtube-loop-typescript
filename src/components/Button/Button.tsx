import React, { FunctionComponent } from 'react'
import './Button.scss'

type ButtonProps = {
  className?: string,
  color?: string
  onClick?: () => void
}

export const Button: FunctionComponent<ButtonProps> = ({ className = '', color, onClick, children }) => {
  let buttonClass = className ? className + ' button' : 'button'

  if (color) {
    buttonClass += ` button--${color}`
  }

  return (
    <button className={buttonClass} onClick={onClick}>
      {children}
    </button>
  )
}

export default Button