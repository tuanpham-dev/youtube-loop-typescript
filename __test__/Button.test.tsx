import React from 'react'
import TestRenderer from 'react-test-renderer'
import Button from '../src/components/Button/Button'

describe('Button', () => {
  it('should render normal button', () => {
    const button = TestRenderer.create(
      <Button>Normal Button</Button>
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it('should render color button', () => {
    const blueButton = TestRenderer.create(
      <Button color="blue">Blue Button</Button>
    ).toJSON()
    expect(blueButton).toMatchSnapshot()

    const redButton = TestRenderer.create(
      <Button color="red">Red Button</Button>
    ).toJSON()
    expect(redButton).toMatchSnapshot()
  })

  it('should handle onClick function', () => {
    const button = TestRenderer.create(
      <Button onClick={() => console.log('clicked')}>onClick Button</Button>
    ).toJSON()
    expect(button).toMatchSnapshot()
  })
})