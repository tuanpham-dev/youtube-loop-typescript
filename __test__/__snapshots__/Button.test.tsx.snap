// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should handle onClick function 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  onClick Button
</button>
`;

exports[`Button should render color button 1`] = `
<button
  className="button button--blue"
>
  Blue Button
</button>
`;

exports[`Button should render color button 2`] = `
<button
  className="button button--red"
>
  Red Button
</button>
`;

exports[`Button should render normal button 1`] = `
<button
  className="button"
>
  Normal Button
</button>
`;
